docs/source/links/rest.md

```diff
@@ -0,0 +1,148 @@
+
+<h2 id="context">Context</h2>
+
+REST Link uses the `headers` field on the context to allow passing headers to the HTTP request. It also supports the `credentials` field for defining credentials policy.
+
+- `headers`: an object representing values to be sent as headers on the request
+- `credentials`: a string representing the credentials policy you want for the fetch call
+
+<h2 id="query">Writing your first query</h2>
+
+To write your first query, add a `@rest` directive to the field that should be resolved from your rest endpoint:
+
+```js
+const query = gql`
+  query example {
+    repos @rest(type: "Repo", path: "/repos") {
+      id
+      name
+    }
+  }
+`;
+```
+
+<h2 id="rest">@rest directive</h2>
+
+This is where you setup the endpoint you want to fetch.
+The rest directive could be used at any depth in a query, but once it is used, nothing nested in it can be GraphQL data, it has to be from the rest link or other resource (like a @client directive)
+
+<h3 id="export.arguments">Arguments</h3>
+
+A `@rest` directive takes three arguments:
+  - path: path to rest endpoint. This could be a path or a full url. If a path, add to the endpoint given on link creation or from the context is concatenated to it.
+  - method: the HTTP method to send the request via (i.e GET, PUT, POST)
+  - type: The GraphQL type this will return
+  - endpoint: which endpoint (if using a map of endpoints) to use for the request
+
+<h3 id="export.variables">Variables</h3>
+
+You can use query variables inside the path argument of your directive for example
+
+```graphql
+query postTitle {
+  post(id: "1") @rest(type: "Post", path: "/post/:id") {
+    id
+    title
+  }
+}
+```
+
+<h2 id="export">@export directive</h2>
+
+The export directive re-exposes a field for use in a later (nested) query. An example use-case would be getting a list of users, and hitting a different endpoint to fetch more data using the exported field in the REST query args.
+
+<h3 id="export.arguments">Arguments</h3>
+  - as: the string name to create this as a variable to be used down the selection set
+
+<h3 id="export.notes">Notes</h3>
+
+These are the same semantics that will be supported on the server, but when used in a rest link you can use the exported variables for further calls (i.e. waterfall requests from nested fields)
+
+```graphql
+const QUERY = gql`
+  query RestData($email: String!) {
+    users @rest(path: '/users/email/:email', params: { email: $email }, method: 'GET', type: 'User') {
+      id @export(as: "id")
+      firstName
+      lastName
+      friends @rest(path: '/friends/:id', params: { id: $id }, type: '[User]') {
+        firstName
+        lastName
+      }
+    }
+  }
+`;
+```
+
+<h2 id="mutation">Mutations</h2>
+
+You can write also mutations with `apollo-link-rest`, for example:
+
+```graphql
+const createPostMutation = gql`
+  fragment PublishablePostInput on REST {
```

Comment:
Why are we using a fragment here? REST isn't a defined type, so this seems confusing

Reply:
I simlified the example